hyperspy/signal.py

```diff
@@ -3867,6 +3867,157 @@ def add_gaussian_noise(self, std):
+                signal_axes = tuple(am[ax] for ax in signal_axes)
+                navigation_axes = tuple(ax for ax in nat_axes if ax not in
+                                        signal_axes)
+            elif iterable_not_string(navigation_axes):
+                # want to keep the order
+                signal_axes = tuple(am[ax] for ax in signal_axes)
+                navigation_axes = tuple(am[ax] for ax in navigation_axes)
+                if len(set(signal_axes).intersection(navigation_axes)):
+                    raise ValueError("Same axis in both spaces found.")
+                if len(am._axes) != (len(signal_axes) + len(navigation_axes)):
+                    raise ValueError("Not all current axes were assigned to a "
+                                     "space")
+            else:
+                raise ValueError("navigation_axes has to be None or an iterable"
+                                 " when signal_axes is iterable")
+        elif signal_axes is None:
+            if isinstance(navigation_axes, int):
+                if len(nat_axes) < navigation_axes:
+                    raise ValueError("Too many navigation axes requested")
+                signal_axes = nat_axes[navigation_axes:]
+                navigation_axes = nat_axes[:navigation_axes]
+            elif iterable_not_string(navigation_axes):
+                navigation_axes = tuple(am[ax] for ax in navigation_axes)
+                signal_axes = tuple(ax for ax in nat_axes if ax not in
+                                    navigation_axes)
+            elif navigation_axes is None:
+                signal_axes = am.navigation_axes
+                navigation_axes = am.signal_axes
+            else:
+                raise ValueError("The passed navigation_axes argument is not valid")
+        else:
+            raise ValueError("The passed signal_axes argument is not valid")
+        # get data view
+        array_order = tuple(
+            ax.index_in_array for ax in reversed(navigation_axes))
+        array_order += tuple(ax.index_in_array for ax in reversed(signal_axes))
+        newdata = self.data.transpose(array_order)
+        res = self._deepcopy_with_new_data(newdata)
+
+        # reconfigure the axes of the axesmanager:
+        ram = res.axes_manager
+        ram._update_trait_handlers(remove=True)
+        # _axes are ordered in array order
+        ram._axes = [ram._axes[i] for i in array_order]
+        for i, ax in enumerate(ram._axes):
+            if i < len(navigation_axes):
+                ax.slice = None
+                ax.navigate = True
+            else:
+                ax.slice = slice(None)
+                ax.navigate = False
+        ram._update_attributes()
+        ram._update_trait_handlers(remove=False)
+        # need a copy?
+        if copy:
+            res._make_sure_data_is_contiguous()
+        return res
+
+    @contextmanager
+    def transposed(self, signal_axes=None, navigation_axes=None, copy=False):
```

Comment:
The context manager is not very useful because transpose returns the signal in a "different view" rather than modifying it. Therefore the context manager has nothing to do when exiting. (My bad, as I suggested the CM to you.)